src/uct/ib/rc/accel/rc_mlx5_devx.c

```diff
@@ -123,26 +124,33 @@ uct_rc_mlx5_iface_common_devx_connect_qp(uct_rc_mlx5_iface_common_t *iface,
     UCT_IB_MLX5DV_SET(init2rtr_qp_in, in_2rtr, opt_param_mask, 14);
 
     qpc = UCT_IB_MLX5DV_ADDR_OF(init2rtr_qp_in, in_2rtr, qpc);
-    UCT_IB_MLX5DV_SET(qpc, qpc, mtu, 5);
+    UCT_IB_MLX5DV_SET(qpc, qpc, mtu, iface->super.config.path_mtu);
     UCT_IB_MLX5DV_SET(qpc, qpc, log_msg_max, 30);
     UCT_IB_MLX5DV_SET(qpc, qpc, remote_qpn, dest_qp_num);
     if (uct_ib_iface_is_roce(&iface->super.super)) {
-        /* TODO resolve L3->L2 address */
-        mac[0] = gid[8] ^ 0x02;
-        memcpy(mac + 1, gid + 9, 2);
-        memcpy(mac + 3, gid + 13, 3);
-        memcpy(UCT_IB_MLX5DV_ADDR_OF(qpc, qpc, primary_address_path.rmac_47_32), mac, 6);
-        UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.hop_limit, 1);
+        status = uct_ib_iface_create_ah(&iface->super.super, ah_attr, &ah);
+        if (status != UCS_OK) {
+            return status;
+        }
+
+        uct_ib_mlx5_get_av(ah, &mlx5_av);
+        memcpy(UCT_IB_MLX5DV_ADDR_OF(qpc, qpc, primary_address_path.rmac_47_32),
+               &mlx5_av.rmac, sizeof(mlx5_av.rmac));
+        memcpy(UCT_IB_MLX5DV_ADDR_OF(qpc, qpc, primary_address_path.rgid_rip),
+               &mlx5_av.rgid, sizeof(mlx5_av.rgid));
+        UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.hop_limit, mlx5_av.hop_limit);
+        UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.src_addr_index, ah_attr->grh.sgid_index);
+        UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.udp_sport, 0xC000 + rand() % 0x4000);
```

Comment:
why do you use this value? Maybe reuse sport assignment from `uct_ib_mlx5_iface_set_av_sport`?